src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java

```diff
@@ -12,11 +16,29 @@ public MovieDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Movie add(Movie movie) {
-        return null;
+        Transaction transaction = null;
+        try (Session session = factory.openSession()) {
+            transaction = session.beginTransaction();
+            session.persist(movie);
+            transaction.commit();
+        } catch (HibernateException e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can`t add movie "
+                    + movie.getTitle() + " to DB. " + e.getMessage());
```

Comment:
Consider using `e` instead of `e.getMessage()` in the exception message. This will provide the full stack trace of the exception, which can be more helpful for debugging purposes.